server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java

```diff
@@ -500,7 +501,13 @@ private void createIndex(SystemIndexMigrationInfo migrationInfo, ActionListener<
         createRequest.waitForActiveShards(ActiveShardCount.ALL)
             .mappings(migrationInfo.getMappings())
             .settings(Objects.requireNonNullElse(settingsBuilder.build(), Settings.EMPTY));
-        metadataCreateIndexService.createIndex(createRequest, listener);
+        metadataCreateIndexService.createIndex(
+            MasterNodeRequest.INFINITE_MASTER_NODE_TIMEOUT,
+            TimeValue.ZERO,
+            null,
```

Comment:
Previously these timeouts were all `null`, but we're trying to eliminate `null` for `masterNodeTimeout` and `ackTimeout` everywhere so this commit replaces them with their non-null equivalents, respectively infinite and zero. It preserves the `null` for `waitForActiveShardsTimeout` tho.

Reply:
For `ackTimeout`, `null` maps to `ZERO` because of this code:

https://github.com/elastic/elasticsearch/blob/a59c182f9f7e9d1bf3d6eecbc0e44f24ff91d053/server/src/main/java/org/elasticsearch/cluster/service/MasterService.java#L721-L725

but then `timeLeft` will always be `0L` a few lines down, so we do indeed immediately call `onTimeout()`.